moonstone/plot/counts.py

```diff
@@ -35,51 +39,83 @@ def plot_mean_distribution(
+                plotting_options, "xaxes", "tickangle", -60
+            )
 
         mean_series = self.df.mean(axis=1)
         binned_mean = SeriesBinning(mean_series).binned_data
-        bar_fig = BarGraph(binned_mean, plotting_options, show=show, output_file=output_file)
-        bar_fig.plot_one_graph(
-            plotting_options,
-            show=show,
-            output_file=output_file
+        bar_fig = BarGraph(
+            binned_mean, plotting_options, show=show, output_file=output_file
         )
+        bar_fig.plot_one_graph(plotting_options, show=show, output_file=output_file)
 
 
-# class PlotTaxonomyStats():
-#     """
-#     Several plots available to visualize taxonomy count data.
-#     """
-
-#     def __init__(self, dataframe: pd.DataFrame, items_name: str = "items"):
-#         self.df = dataframe
-#         self.items_name = items_name
-
-#     def _plot_taxonomy_classification(self, level_of_interest, plotting_options: dict = None,
-#                                       show: Optional[bool] = True, output_file: Optional[str] = False):
-#         """
-#         ~~~~ IN CONSTRUCTION (remove _ in title when finished) ~~~~
-#         method to visualize the fractions of taxonomic levels (species/genus/...)
-
-#         :param level_of_interest: taxonomic classification level (species, genus etc.) to plot
-#         :param output_file: name of the output file
-#         :param plotting_options: options of plotting that will override the default setup \n
-#                                  [!] Make sure the value given to an argument is of the right type \n
-#                                  options allowed : 'log': `bool` ; 'colorbar': `[str, List[str]]` ;
-#                                  'tickangle': `[int, float]`
-#         """
+class PlotTaxonomyCounts:
+    """
+    Plots available for taxonomy counts (multiindexed dataframe).
+    """
 
-        # mean by sample or nb of samples with ?
+    def __init__(self, taxonomy_dataframe: pd.DataFrame):
+        self.df = taxonomy_dataframe
 
-    # heatmap -> in analysis?
+    def plot_most_abundant_taxa(
+        self,
+        mean_taxa: float = None,
+        taxa_number: int = 20,
+        taxa_level: str = "species",
+        **kwargs,
+    ):
+        data_df = self.df
+        if mean_taxa is not None:
+            data_df = TaxonomyMeanFiltering(data_df, mean_taxa).filtered_df
+        # Select top `taxa_number`
+        top_taxa = (
+            data_df.groupby(taxa_level)
+            .sum()
+            .sum(axis=1)
+            .sort_values(ascending=False)[:taxa_number]
+            .index
+        )
+        number_of_taxa = len(
+            top_taxa
+        )  # Can be different than threshold if not enough taxa
+        # Filter for top species
+        abundances = data_df.groupby(taxa_level).sum().loc[top_taxa]
+        percentage_presence = (abundances != 0).sum(axis=1) / abundances.shape[1] * 100
```

Comment:
Here, you're recalculating the percentage only among the top n species, without "Others", right? (And just by curiosity, why do you use (abundances != 0) instead of just abundances?)